File: frontend/src/types/conversation.ts
// 对话和消息相关类型定义

export interface Conversation {
  id: string
  user_id: string
  title: string
  description?: string
  is_archived: boolean
  created_at: string
  updated_at: string
  message_count?: number
  last_message_at?: string
  username?: string
  full_name?: string
}

export interface Message {
  id: string
  conversation_id: string
  role: 'user' | 'assistant' | 'system'
  content: string
  tokens_used: number
  metadata?: any
  created_at: string
}

export interface CreateConversationData {
  title?: string
  description?: string
}

export interface SendMessageData {
  content: string
  conversation_id?: string  // 如果为空，创建新对话
}

export interface ConversationResponse {
  conversation: Conversation
  messages: Message[]
}

export interface SearchConversationParams {
  page?: number
  page_size?: number
  keyword?: string
  is_archived?: boolean
}

File: backend/app/models/conversation.py
"""
对话和消息数据模型
"""

from sqlalchemy import Column, String, Text, Integer, Boolean, DateTime, ForeignKey, JSON
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
import uuid

from ..core.database import Base

class Conversation(Base):
    """对话会话模型"""
    __tablename__ = "conversations"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    user_id = Column(UUID(as_uuid=True), ForeignKey("users.id"), nullable=False, index=True)
    title = Column(String(200), nullable=False, default="新对话")
    description = Column(Text)
    is_archived = Column(Boolean, default=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())

    # 关系定义
    user = relationship("User", back_populates="conversations")
    messages = relationship("Message", back_populates="conversation", cascade="all, delete-orphan")

    def __repr__(self):
        return f"<Conversation(id={self.id}, title={self.title}, user_id={self.user_id})>"


class Message(Base):
    """消息模型"""
    __tablename__ = "messages"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    conversation_id = Column(UUID(as_uuid=True), ForeignKey("conversations.id"), nullable=False, index=True)
    role = Column(String(20), nullable=False)  # user, assistant, system
    content = Column(Text, nullable=False)
    tokens_used = Column(Integer, default=0)
    meta_data = Column(JSON)  # 存储额外信息，如引用的文档等
    created_at = Column(DateTime(timezone=True), server_default=func.now())

    # 关系定义
    conversation = relationship("Conversation", back_populates="messages")

    def __repr__(self):
        return f"<Message(id={self.id}, role={self.role}, conversation_id={self.conversation_id})>"